frontend/public/components/utils/name-value-editor.jsx

```diff
@@ -58,7 +58,9 @@ export const NameValueEditor = withDragDropContext(class NameValueEditor extends
   }
 
   render () {
-    const {nameString, valueString, addString, nameValuePairs, allowSorting, readOnly, nameValueId, configMaps, secrets, envFrom} = this.props;
+    const {nameString, valueString, addString, nameValuePairs, allowSorting, readOnly, nameValueId, configMaps, secrets, envFrom, isCreate} = this.props;
+    //Don't show Add for config maps/secrets unless we are allowing sorting (env editor not annotations) but not if we are on a create page or envFrom
+    const showAddConfigMapSecret = allowSorting && !isCreate && !envFrom;
```

Comment:
This should be an explicit prop rather than guessing based on values like `allowSorting`.